downloader/downloader.go

```diff
@@ -287,32 +357,39 @@ func (downloader *Downloader) acceptSpace(result *downloadResult) (err error) {
 	return err
 }
 
-func (downloader *Downloader) setDownloadResultOutdated(result *downloadResult) (err error) {
-	if setOutdatedErr := downloader.setItemOutdated(result.downloadFileName); setOutdatedErr != nil && err == nil {
-		err = setOutdatedErr
-	}
-
-	if setOutdatedErr := downloader.setItemOutdated(result.interruptFileName); setOutdatedErr != nil && err == nil {
-		err = setOutdatedErr
+func (downloader *Downloader) setDownloadDirOutdated() error {
+	downloadInfos, err := downloader.storage.GetDownloadInfos()
+	if err != nil {
+		return aoserrors.Wrap(err)
 	}
 
-	return err
-}
-
-func (downloader *Downloader) setDownloadDirOutdated() (err error) {
 	entries, err := os.ReadDir(downloader.config.DownloadDir)
 	if err != nil {
 		return aoserrors.Wrap(err)
 	}
 
+nextEntry:
 	for _, entry := range entries {
-		if setOutdatedErr := downloader.setItemOutdated(
-			filepath.Join(downloader.config.DownloadDir, entry.Name())); setOutdatedErr != nil && err == nil {
-			err = setOutdatedErr
+		downloadFilePath := filepath.Join(downloader.config.DownloadDir, entry.Name())
+
+		for _, downloadInfo := range downloadInfos {
+			if !downloadInfo.Downloaded {
```

Comment:
Why do you need this check on each iteration? What's wrong with proposed code:
```go
	for _, entry := range entries {
		downloadFilePath := filepath.Join(downloader.config.DownloadDir, entry.Name())

		for _, downloadInfo := range downloadInfos {
			if downloadFilePath == downloadInfo.Path && donwloadInfo.Downloaded {
				continue nextEntry
			}
		}

                if err := downloader.setItemOutdated(downloadFilePath); err != nil {
                       ...
                }
	}
```

Reply:
Done